### ISSUE TITLE
Scan all the containers in a Pod

### ISSUE THREAD

**Issue by luarx** [controlplaneio/kubectl-kubesec#10]
When I use `kubectl -n default plugin scan deployment/[name]` it only scans 1 container and **it doesn't scan the rest**.

EXAMPLE

- Deployment with 2 containers. One of them satifies the security features:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: 'deployment-name'
  labels:
    app: 'deployment-name'
spec:
  replicas: 2
  selector:
    matchLabels:
      app: 'deployment-name'
  template:
    metadata:
      labels:
        app: 'deployment-name'
    spec:
      containers:
        - name: container
          image: repo/name:tag
          securityContext:
            runAsUser: 10001
            runAsGroup: 10001
            runAsNonRoot: true
            capabilities:
              drop: ['ALL']
            readOnlyRootFilesystem: true
          resources:
            requests:
              memory: '400Mi'
              cpu: '100m'
            limits:
              memory: '500Mi'
              cpu: '200m'

        - name: container2
          image: repo/name:tag
          resources:
            requests:
              memory: "100Mi"
              cpu: "50m"
            limits:
              memory: "200Mi"
              cpu: "100m"
```
- RESULT:
```
{
  "score": 9,
  "scoring": {
    "critical": [],
    "advise": []
  }
}
```

- Same deployment without the container that satifies the security advises:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: 'deployment-name'
  labels:
    app: 'deployment-name'
spec:
  replicas: 2
  selector:
    matchLabels:
      app: 'deployment-name'
  template:
    metadata:
      labels:
        app: 'deployment-name'
    spec:
      containers:
        - name: container2
          image: repo/name:tag
          resources:
            requests:
              memory: "100Mi"
              cpu: "50m"
            limits:
              memory: "200Mi"
              cpu: "100m"
```

- RESULT:
```
{
  "score": 4,
  "scoring": {
    "critical": [],
    "advise": [
      {
        "selector": "containers[] .securityContext .runAsNonRoot == true",
        "reason": "Force the running image to run as a non-root user to ensure least privilege"
      },
      {
        "selector": "containers[] .securityContext .capabilities .drop",
        "reason": "Reducing kernel capabilities available to a container limits its attack surface",
        "href": "https://kubernetes.io/docs/tasks/configure-pod-container/security-context/"
      },
      {
        "selector": "containers[] .securityContext .readOnlyRootFilesystem == true",
        "reason": "An immutable root filesystem can prevent malicious binaries being added to PATH and increase attack cost"
      },
      {
        "selector": "containers[] .securityContext .runAsUser > 10000",
        "reason": "Run as a high-UID user to avoid conflicts with the host's user table"
      },
      {
        "selector": "containers[] .securityContext .capabilities .drop | index(\"ALL\")",
        "reason": "Drop all capabilities and add only those required to reduce syscall attack surface"
      }
    ]
  }
}
```

Is it normal? Is there an option so that I can scan all the containers in a POD?

Thank you very much for this helpful tool!

**Comment by RealArtemiy** [comment]
On one deployment all containers are same

**Comment by luarx** [comment]
Sorry, but I don't understand your answer...

If I have 2 containers and one of them is not secure, why if I have both containers defined in my deployment it says that it's good and if I delete 1 of them it says that it's not secure? :O 
If they are the same it should be secure in every situation, right?

**Comment by luarx** [comment]
Any update on this?

**Comment by sublimino** [comment]
Hi @luarx, kubesec V2 now scores for all containers. The plugin hasn't been updated yet, but you can try it in the browser at kubesec.io

**Comment by luarx** [comment]
Thanks!!!
I would try to use it locally, do you know when the plugin will be updated?

**Comment by spiarh** [comment]
There is a new release with a bunch of fixes and that includes this (old) feature :wink: 

Release: https://github.com/controlplaneio/kubectl-kubesec/releases/tag/v1.1.0
